sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/SettingFieldsExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.Data.AppConfiguration
+{
+    internal static class SettingFieldsExtensions
+    {
+        /// <summary>
+        /// Maps a SettingFields member to its corresponding service name in accordance with the REST API specification.
+        /// </summary>
+        private static readonly IReadOnlyDictionary<SettingFields, string> s_serviceNameMap = new Dictionary<SettingFields, string>()
+        {
+            { SettingFields.Key         , "key"           },
+            { SettingFields.Label       , "label"         },
+            { SettingFields.Value       , "value"         },
+            { SettingFields.ContentType , "content_type"  },
+            { SettingFields.ETag        , "etag"          },
+            { SettingFields.LastModified, "last_modified" },
+            { SettingFields.IsReadOnly  , "locked"        },
+            { SettingFields.Tags        , "tags"          }
+        };
```

Comment:
I preferred creating an explicit mapping of service names to make it less error-prone. In case the service ever adds a new member to the `SettingFields` enum, I have a test in place to fail so we don't forget to update this dictionary.